src/l500/l500-color.cpp

```diff
@@ -221,7 +244,75 @@ namespace librealsense
                 return intrinsics;
             }
         }
-        throw std::runtime_error( to_string() << "intrinsics for resolution " << profile.width << "," << profile.height << " don't exist" );
+        throw std::runtime_error( to_string() << "intrinsics for resolution " << width << ","
+                                              << height << " don't exist" );
+    }
+
+    double l500_color_sensor::read_temperature() const
+    {
+            auto hwm = *_owner->_hw_monitor;
```

Comment:
I believe this will copy... maybe store a &?

Reply:
done